include/shad/core/impl/minimum_maximum_ops.h

```diff
@@ -0,0 +1,615 @@
+  using itr_traits = distributed_iterator_traits<ForwardIt>;
+  auto localities = itr_traits::localities(first, last);
+  using value_t =
+    typename std::remove_const<typename itr_traits::value_type>::type;
+  using res_t = std::pair<ForwardIt, value_t>;
+  auto args = std::make_tuple(first, last, comp);
+  std::vector<res_t> res(localities.size());
+  size_t i = 0;
+  for (auto locality = localities.begin(), end = localities.end();
+       locality != end; ++locality, ++i) {
+    rt::executeAtWithRet(
+        locality,
+        [](const std::tuple<ForwardIt, ForwardIt, Compare>& args,
+           res_t* result) {
+          auto gbegin = std::get<0>(args);
+          auto gend = std::get<1>(args);
+          auto local_range = itr_traits::local_range(gbegin, gend);
+          auto begin = local_range.begin();
+          auto end = local_range.end();
+          auto lmax = std::max_element(begin, end, std::get<2>(args));
+          ForwardIt gres = itr_traits::iterator_from_local(gbegin, gend, lmax);
+          *result = std::make_pair(gres, *lmax);
+        },
+        args, &res[i]);
+  }
+  auto it = res.begin();
+  ForwardIt result = it->first;
+  value_t max = it->second;
+  for (++it; it != res.end(); ++it) {
+    if (comp(max, it->second)) {
+      result = it->first;
+      max = it->second;
+    }
+  }
+  return result;
+}
+
+template <class ForwardIt, class Compare>
+ForwardIt max_element(distributed_parallel_tag&& policy,
+                      ForwardIt first, ForwardIt last, Compare comp) {
+  if (first == last) return last;
+  using itr_traits = distributed_iterator_traits<ForwardIt>;
+  auto localities = itr_traits::localities(first, last);
+  using value_t =
+    typename std::remove_const<typename itr_traits::value_type>::type;
+  using res_t = std::pair<ForwardIt, value_t>;
+  std::vector<res_t> res(localities.size());
+  size_t i = 0;
+  rt::Handle h;
+  auto args = std::make_tuple(first, last, comp);
+  for (auto locality = localities.begin(), end = localities.end();
+       locality != end; ++locality, ++i) {
+    rt::asyncExecuteAtWithRet(
+      h, locality,
+      [](rt::Handle&,
+         const std::tuple<ForwardIt, ForwardIt, Compare>& args,
+         res_t* result) {
+          auto gbegin = std::get<0>(args);
+          auto gend = std::get<1>(args);
+          auto local_range = itr_traits::local_range(gbegin, gend);
+          auto begin = local_range.begin();
+          auto end = local_range.end();
+          auto lmax = std::max_element(begin, end, std::get<2>(args));
+          ForwardIt gres = itr_traits::iterator_from_local(gbegin, gend, lmax);
+          *result = std::make_pair(gres, *lmax);
+      },
+      args, &res[i]);
+  }
+  rt::waitForCompletion(h);
+  auto it = res.begin();
+  ForwardIt result = it->first;
+  value_t max = it->second;
+  for (++it; it != res.end(); ++it) {
+    if (comp(max, it->second)) {
+      result = it->first;
+      max = it->second;
+    }
+  }
+  return result;
+}
+
```

Comment:
Why not expressing the cases with operator< comparison as special cases of the Compare ones?